### ISSUE TITLE
Dynamic Linking of SpiderMonkey Interpreter

### ISSUE THREAD

**Issue by royaljain** [bytecodealliance/componentize-js#66]
**Problem**

The size of wasm module produced is quite high, ~ 8 MB for a function adding two numbers. My understanding is that the interpreter (SpiderMonkey) is the majority of it.

**Query**

Is there a way to dynamically link the interpreter and wasm module to reduce the module size? The primary benefit is when there are multiple wasm modules & instances

For reference: [Javy](https://shopify.engineering/javascript-in-webassembly-for-shopify-functions) has -d flag for enabling dynamic linking

**Comment by guybedford** [comment]
Yes this remains a goal for this project, where the engine itself can be imported as a core module that can be dynamically linked.

This work is not currently being prioritised for a while yet, unless someone wants to pick it up.

**Comment by royaljain** [comment]
Got it.

Is this connected to the component model, will the linking be done through that?

**Comment by maxbrunsfeld** [comment]
I just ran into this as well. I'm seeing `componentize()` produce a 10MB binary when I give it a module with nothing but empty stub functions for my [world](https://github.com/zed-industries/zed/tree/main/crates/extension_api/wit/since_v0.2.0).
